src/responder/common/cache_req/cache_req_private.h

```diff
@@ -68,6 +68,7 @@ struct cache_req_parsed_name {
     const char *input;  /* Original input. */
     const char *name;   /* Parsed name or UPN. */
     const char *lookup; /* Converted per domain rules. */
+    const char *attr;   /* Attribute name when looking for an attribute */
```

Comment:
IMO this look a bit strange to me, is there an actual `name` provided here?
~~~
    input.name.attr = attr;
~~~
I don't insist on it but I wonder if attr can or should be moved to be part of `struct cache_req_data`

Reply:
It's probably okay to leave it how you have it, but can we improve the comment for this struct.

/**
 * Structure to hold the input strings that
 * should be parsed into name and domain parts.
 */

Reply:
You are right in that `name` is not used when the user calls `ListByAttr()`, but there is one when the user calls some other functions.

I think `name.attr` should be part of the `cache_req_parsed_name` structure because `name.attr` and `name.name` are both used by different functions; together with the `name.input` and `name.lookup` fields. But they cannot be merger into a single field or a `union` because some intermediate function will set a value in `name.name` overwriting the value of `name.attr` (if they were merged).

As an intermediate solution we could just rename the above level (also called `name`) from `name` to something else, for instance `params` (or anything else), so that the example you gave above would become `input.params.attr`. Is that OK for you?

Reply:
Oh. Your second comment arrived while I was writing the answer and didn't see it in time.

Reply:
Thank you.